src/main/scala/scalaz/parsers/ParserSyntax.scala

```diff
@@ -33,11 +33,11 @@ trait ParserSyntax[P[_], F[_], G[_]] {
     def || (other: => P[A])(implicit A: Alternative[P]): P[A] =
       A.or(p, other)
 
-    def ∘ [B](instance: iso.Iso[A, B]): P[B] =
-      isoMap(p)(instance)
+    def ∘ [B](instance: parsing.Equiv[A, B]): P[B] =
+      imap(p)(instance)
 
     def many(implicit P: ProductFunctor[P], A: Alternative[P]): P[List[A]] = {
-      lazy val step: P[List[A]] = (lift(()) \/ (p /\ delay(step))) ∘ iso.list
+      lazy val step: P[List[A]] = ((p /\ delay(step)) \/ lift(())) ∘ parsing.Equiv.list
```

Comment:
what is the reason for moving `lift(())` to the right side of the `\/`?  Also, can we use `P.unit` in place of `lift(())`

Reply:
> what is the reason for moving lift(()) to the right side of the \/?

Because `lift` always succeeds.

> Also, can we use P.unit in place of lift(())

```scala
def lift[A](a: A): P[A]
```
lifts an `A` into `P`, there is no `unit`